webpack/assets/javascripts/react_app/routes/index.js

```diff
@@ -1,43 +1,70 @@
 import React from 'react';
 import { Switch, Route } from 'react-router-dom';
+import URI from 'urijs';
 import { routes } from './routes';
 
-let currentLocation = null;
-
-const AppSwitcher = () => (
-  <Switch>
-    {routes.map(({ render, path, ...routeProps }) => (
-      <Route
-        path={path}
-        key={path}
-        {...routeProps}
-        render={renderProps => {
-          const railsContainer = document.getElementById('rails-app-content');
-          if (railsContainer) railsContainer.remove();
-          currentLocation = renderProps.location;
-
-          return render(renderProps);
-        }}
-      />
-    ))}
-    <Route
-      render={child => {
-        if (
-          currentLocation &&
-          currentLocation.pathname !== child.location.pathname
-        ) {
-          const useTurbolinks =
-            child.location.state &&
-            child.location.state.useTurbolinks &&
-            !window.history.state.turbolinks; // visit() already called
-
-          if (useTurbolinks) window.Turbolinks.visit(child.location.pathname);
-        }
-        currentLocation = child.location;
-        return null;
-      }}
-    />
-  </Switch>
-);
+let currentPath = null;
+
+const AppSwitcher = () => {
+  const updateCurrentPath = () => {
+    currentPath = getURIPath();
+  };
+
+  const getURIPath = () => {
+    const uri = new URI();
+    return uri.pathname();
+  };
```

Comment:
can you please add a comment (and to our tech debt) above that this is a workaround due to turbolinks location manipulation?

Reply:
Thanks, added :+1: